### DISCUSSION TITLE
Should asyncIterator usage of stream be documented?

### DISCUSSION THREAD

**Discussion by ejose19** [sindresorhus/got#1772] [Ideas]
I'm aware of the event / pipe based  usage, however there's another way of consuming the `ReadableStream` (that may be more simple in some cases):

```typescript
type TypedStream = Request & {
  [Symbol.asyncIterator](): AsyncIterableIterator<Buffer>;
};

const stream: TypedStream = got.stream("https://sindresorhus.com/");

for await (const chunk of stream) {
  // Process each chunk
}
```

In case this would be good to have it documented, where should it be placed? I can submit a PR.

PS: Node `Readable` has `[Symbol.asyncIterator](): AsyncIterableIterator<any>;`, hence the need to retype in order for `chunk` type to be accurate.

**Comment by sindresorhus** [comment]
> In case this would be good to have it documented, where should it be placed? I can submit a PR.

Yeah, sure. But it should probably wait until https://github.com/sindresorhus/got/pull/1699 is merged as it reorganizes the docs.
